provisioner/aws-manager.js

```diff
@@ -70,15 +102,75 @@ AwsManager.prototype.update = function () {
         },
       ],
     }),
+    that.ec2.describeInstances({
+      Filters: [
+        {
+          Name: 'key-name',
+          Values: [that.keyPrefix + '*'],
+        },
+        {
+          Name: 'instance-state-name',
+          Values: ['shutting-down', 'terminated', 'stopping'],
+        },
+      ],
+    }),
+    that.ec2.describeSpotInstanceRequests({
+      Filters: [
+        {
+          Name: 'launch.key-name',
+          Values: [that.keyPrefix + '*'],
+        }, {
+          Name: 'state',
+          Values: ['cancelled', 'failed', 'closed', 'active'],
+        },
+      ],
+    }),
   ]);
 
   p = p.then(function (res) {
-    if (res[0].NextToken && res[0].NextToken !== '') {
+    var livingInstances = res[0];
+    var livingSpotRequests = res[1];
+    var deadInstances = res[2];
+    var deadSpotRequests = {};
+
+    // NOTE: Dead spot request does not mean that the spot request failed,
+    //       just that it's no longer an open spot request...
+
+    // TODO: Probably better to do a single set of describe calls and bucket
+    //       them in this section... hmm, no strong feelings until data
+
+    // Because we don't filter the spot requests, we'll need to make sure that
+    // we remove the SpotInstanceRequests per-region response so that
+    // that._classify() knows how to interpret the data
+    Object.keys(res[3]).forEach(function (region) {
+      deadSpotRequests[region] = res[3][region].SpotInstanceRequests;
+    });
```

Comment:
Curious, why not filter them? If we need additional properties how about we just extend the filter to keep those...
I think it's preferable to have objects for spot requests that are similar form (has same set of properties).